File: sync/src/main/scala/com/fan/mysql/parser/XidEventParser.java
package com.dingcloud.dts.binlog.parser;

import com.dingcloud.dts.binlog.BinlogEventParser;
import com.dingcloud.dts.binlog.event.BinlogEvent;
import com.dingcloud.dts.binlog.event.EventHeader;
import com.dingcloud.dts.binlog.event.impl.XidEvent;
import com.dingcloud.dts.binlog.mysql.dbsync.BinlogContext;
import com.dingcloud.dts.binlog.mysql.dbsync.LogBuffer;

public class XidEventParser implements BinlogEventParser {

	public BinlogEvent parse(LogBuffer buffer, EventHeader header, BinlogContext context) {
		long xid = buffer.getLong64();
		XidEvent event = new XidEvent(header);
		event.setXid(xid);
		context.getTableMapEvents().clear();
		return event;
	}

}

File: async/src/main/scala/com/fan/mysql/async/binlog/parse/XidEventParser.scala
package com.fan.mysql.async.binlog.parse

import com.fan.mysql.async.binlog.event.impl.XidEvent
import com.fan.mysql.async.binlog.event.{BinlogEvent, EventHeader}
import com.fan.mysql.async.binlog.{BinlogDumpContext, BinlogEventParser}
import io.netty.buffer.ByteBuf

class XidEventParser extends BinlogEventParser {

  override def parse(buffer: ByteBuf, header: EventHeader, context: BinlogDumpContext): BinlogEvent = {
    val xid = buffer.readLong()
    val event = new XidEvent(header)
    event.setXid(xid)
    context.getTableMapEvents.clear()
    event
  }
}
